web/models/build_status_updater.ex

```diff
@@ -1,9 +1,27 @@
 defmodule Bigben.BuildStatusUpdater do
+  alias Bigben.Branch
   alias Bigben.Build
   alias Bigben.Repo
 
   def update(build_params) do
-    changeset = Build.changeset(%Build{}, build_params)
+    branch = get_branch(build_params)
+    updated_params = build_params
+      |> Map.delete(:branch)
+      |> Map.put("branch_id", branch.id)
```

Comment:
I believe you can do this without passing a third parameter. You can get at the params given to a changeset with `changeset.params["whatever_the_key_is"]`. That how `validate_confirmation` works